### PR TITLE
Use name for directory/file initializing and removed unwanted funcion…

### PR DESCRIPTION
**Pull request by katmsft**
…ality.

Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/1163

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-shares/src/share_client.cpp [hunk 1]
```diff
@@ -90,21 +90,7 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
...
-    builder.AppendPath(Storage::Details::UrlEncodePath(directoryPath));
-    return ShareDirectoryClient(builder, m_pipeline);
-  }
-
-  ShareFileClient ShareClient::GetFileClient(const std::string& filePath) const
-  {
-    auto builder = m_shareUri;
-    builder.AppendPath(Storage::Details::UrlEncodePath(filePath));
-    return ShareFileClient(builder, m_pipeline);
+    return ShareDirectoryClient(m_shareUri, m_pipeline);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/src/share_client.cpp:93]: You sure we don't need some slash (/) here.

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-shares/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## 12.0.0-beta.7 (Unreleased)
 
+### Breaking Changes
+
+- Removed `GetDirectoryClient` and `GetFileClient` from `ShareClient`. `ShareDirectoryClient` and `ShareFileClient` now initializes with the name of the resource, not path, to indicate that no path parsing is done for the API
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/CHANGELOG.md:7]:
```suggestion
- Removed `GetDirectoryClient` and `GetFileClient` from `ShareClient`. `ShareDirectoryClient` and `ShareFileClient` now initializes with the name of the resource, not path, to indicate that no path parsing is done for the API.
````

**Pull request merged**